src/shell/CMakeLists.txt

```diff
@@ -1,11 +1,16 @@
-add_library(shell OBJECT lean.cpp)
-
-add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/bin/lean${CMAKE_EXECUTABLE_SUFFIX}
-  COMMAND sh -c "LEANC_GMP=${GMP_LIBRARIES} ${CMAKE_BINARY_DIR}/bin/leanc -x none ${CMAKE_EXE_LINKER_FLAGS} $<TARGET_OBJECTS:shell> -o ${CMAKE_BINARY_DIR}/bin/lean${CMAKE_EXECUTABLE_SUFFIX}"
-  DEPENDS Init Std Lean leancpp shell)
-
-add_custom_target(lean ALL
-  DEPENDS ${CMAKE_BINARY_DIR}/bin/lean${CMAKE_EXECUTABLE_SUFFIX})
+if (${CMAKE_SYSTEM_NAME} MATCHES "Emscripten")
+  add_executable(lean lean.cpp)
+  set_target_properties(lean PROPERTIES
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin/")
+  target_link_libraries(lean Init Std Lean leancpp "${GMP_INSTALL_PREFIX}/lib/libgmp.a")
```

Comment:
I'm surprised this works. The reason we started linking `lean` using `leanc` was that CMake's support for cyclically dependent libraries was insufficient. I guess WASM linking is just inherently different?

Reply:
It is different, whether sufficiently so to avoid cycle problems is unclear, but so far I have not run into issues with this approach. If need be I can revert to a custom command later.